### ISSUE TITLE
500 error on load

### ISSUE THREAD

**Issue by awesomebing1** [CachetHQ/Cachet#1785]
Before submitting your issue, please make sure that you've checked all of the checkboxes below.

- [x] You're running the [latest release](https://github.com/CachetHQ/Cachet/releases/latest) version of Cachet.
- [x] Ensure that you're running at least PHP 5.5.9, you can check this by running `php -v`
- [x] You've ran `rm -rf bootstrap/cache/*` from the root of your Cachet installation.

To help us better understand your issue, please answer the following — cheers!

### Your setup

- *What version of Cachet?* v2.2.2
- *What database driver? MySQL? Postgres? SQLite?* MySQL
- *What version of PHP?* PHP 7
- *What OS?* Ubuntu Xenial Xerus (16.04)

### Expected behaviour

*Please describe what you're expecting to see happen.*

I setup Cachet.

### Actual behaviour

*Please describe what you're actually seeing happen.*

```
[Fri May 13 23:49:26.840307 2016] [:error] [pid 20412] [client blahblahblah:55221] PHP Fatal error:  Uncaught Illuminate\\Contracts\\Container\\BindingResolutionException: Unresolvable dependency resolving [Parameter #0 [ <required> $path ]] in class GrahamCampbell\\Exceptions\\ExceptionInfo in /var/www/Cachet/vendor/laravel/framework/src/Illuminate/Container/Container.php:839\nStack trace:\n#0 /var/www/Cachet/vendor/laravel/framework/src/Illuminate/Container/Container.php(806): Illuminate\\Container\\Container->resolveNonClass(Object(ReflectionParameter))\n#1 /var/www/Cachet/vendor/laravel/framework/src/Illuminate/Container/Container.php(777): Illuminate\\Container\\Container->getDependencies(Array, Array)\n#2 /var/www/Cachet/vendor/laravel/framework/src/Illuminate/Container/Container.php(633): Illuminate\\Container\\Container->build('GrahamCampbell\\\\...', Array)\n#3 /var/www/Cachet/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(697): Illuminate\\Container\\Container->make('GrahamCampbell\\\\...', Array)\n#4 /var/www/Cachet/vendor/laravel/framework/src/Illuminate/Container/Container.php(853): Illuminate\\Foun in /var/www/Cachet/vendor/laravel/framework/src/Illuminate/Container/Container.php on line 839
[Fri May 13 23:51:49.689572 2016] [:error] [pid 20387] [client blahblahblah:55298] PHP Fatal error:  Uncaught Illuminate\\Contracts\\Container\\BindingResolutionException: Unresolvable dependency resolving [Parameter #0 [ <required> $path ]] in class GrahamCampbell\\Exceptions\\ExceptionInfo in /var/www/Cachet/vendor/laravel/framework/src/Illuminate/Container/Container.php:839\nStack trace:\n#0 /var/www/Cachet/vendor/laravel/framework/src/Illuminate/Container/Container.php(806): Illuminate\\Container\\Container->resolveNonClass(Object(ReflectionParameter))\n#1 /var/www/Cachet/vendor/laravel/framework/src/Illuminate/Container/Container.php(777): Illuminate\\Container\\Container->getDependencies(Array, Array)\n#2 /var/www/Cachet/vendor/laravel/framework/src/Illuminate/Container/Container.php(633): Illuminate\\Container\\Container->build('GrahamCampbell\\\\...', Array)\n#3 /var/www/Cachet/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(697): Illuminate\\Container\\Container->make('GrahamCampbell\\\\...', Array)\n#4 /var/www/Cachet/vendor/laravel/framework/src/Illuminate/Container/Container.php(853): Illuminate\\Foun in /var/www/Cachet/vendor/laravel/framework/src/Illuminate/Container/Container.php on line 839
[Fri May 13 23:51:49.694559 2016] [:error] [pid 20387] [client blahblahblah:55298] PHP Fatal error:  Uncaught Illuminate\\Contracts\\Container\\BindingResolutionException: Unresolvable dependency resolving [Parameter #0 [ <required> $path ]] in class GrahamCampbell\\Exceptions\\ExceptionInfo in /var/www/Cachet/vendor/laravel/framework/src/Illuminate/Container/Container.php:839\nStack trace:\n#0 /var/www/Cachet/vendor/laravel/framework/src/Illuminate/Container/Container.php(806): Illuminate\\Container\\Container->resolveNonClass(Object(ReflectionParameter))\n#1 /var/www/Cachet/vendor/laravel/framework/src/Illuminate/Container/Container.php(777): Illuminate\\Container\\Container->getDependencies(Array, Array)\n#2 /var/www/Cachet/vendor/laravel/framework/src/Illuminate/Container/Container.php(633): Illuminate\\Container\\Container->build('GrahamCampbell\\\\...', Array)\n#3 /var/www/Cachet/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(697): Illuminate\\Container\\Container->make('GrahamCampbell\\\\...', Array)\n#4 /var/www/Cachet/vendor/laravel/framework/src/Illuminate/Container/Container.php(853): Illuminate\\Foun in /var/www/Cachet/vendor/laravel/framework/src/Illuminate/Container/Container.php on line 839
[Fri May 13 23:52:43.555709 2016] [:error] [pid 20389] [client blahblahblah:55325] PHP Fatal error:  Uncaught Illuminate\\Contracts\\Container\\BindingResolutionException: Unresolvable dependency resolving [Parameter #0 [ <required> $path ]] in class GrahamCampbell\\Exceptions\\ExceptionInfo in /var/www/Cachet/vendor/laravel/framework/src/Illuminate/Container/Container.php:839\nStack trace:\n#0 /var/www/Cachet/vendor/laravel/framework/src/Illuminate/Container/Container.php(806): Illuminate\\Container\\Container->resolveNonClass(Object(ReflectionParameter))\n#1 /var/www/Cachet/vendor/laravel/framework/src/Illuminate/Container/Container.php(777): Illuminate\\Container\\Container->getDependencies(Array, Array)\n#2 /var/www/Cachet/vendor/laravel/framework/src/Illuminate/Container/Container.php(633): Illuminate\\Container\\Container->build('GrahamCampbell\\\\...', Array)\n#3 /var/www/Cachet/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(697): Illuminate\\Container\\Container->make('GrahamCampbell\\\\...', Array)\n#4 /var/www/Cachet/vendor/laravel/framework/src/Illuminate/Container/Container.php(853): Illuminate\\Foun in /var/www/Cachet/vendor/laravel/framework/src/Illuminate/Container/Container.php on line 839
[Fri May 13 23:52:43.558673 2016] [:error] [pid 20389] [client blahblahblah:55325] PHP Fatal error:  Uncaught Illuminate\\Contracts\\Container\\BindingResolutionException: Unresolvable dependency resolving [Parameter #0 [ <required> $path ]] in class GrahamCampbell\\Exceptions\\ExceptionInfo in /var/www/Cachet/vendor/laravel/framework/src/Illuminate/Container/Container.php:839\nStack trace:\n#0 /var/www/Cachet/vendor/laravel/framework/src/Illuminate/Container/Container.php(806): Illuminate\\Container\\Container->resolveNonClass(Object(ReflectionParameter))\n#1 /var/www/Cachet/vendor/laravel/framework/src/Illuminate/Container/Container.php(777): Illuminate\\Container\\Container->getDependencies(Array, Array)\n#2 /var/www/Cachet/vendor/laravel/framework/src/Illuminate/Container/Container.php(633): Illuminate\\Container\\Container->build('GrahamCampbell\\\\...', Array)\n#3 /var/www/Cachet/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(697): Illuminate\\Container\\Container->make('GrahamCampbell\\\\...', Array)\n#4 /var/www/Cachet/vendor/laravel/framework/src/Illuminate/Container/Container.php(853): Illuminate\\Foun in /var/www/Cachet/vendor/laravel/framework/src/Illuminate/Container/Container.php on line 839
[Fri May 13 23:52:47.844965 2016] [:error] [pid 20386] [client blahblahblah:55327] PHP Fatal error:  Uncaught Illuminate\\Contracts\\Container\\BindingResolutionException: Unresolvable dependency resolving [Parameter #0 [ <required> $path ]] in class GrahamCampbell\\Exceptions\\ExceptionInfo in /var/www/Cachet/vendor/laravel/framework/src/Illuminate/Container/Container.php:839\nStack trace:\n#0 /var/www/Cachet/vendor/laravel/framework/src/Illuminate/Container/Container.php(806): Illuminate\\Container\\Container->resolveNonClass(Object(ReflectionParameter))\n#1 /var/www/Cachet/vendor/laravel/framework/src/Illuminate/Container/Container.php(777): Illuminate\\Container\\Container->getDependencies(Array, Array)\n#2 /var/www/Cachet/vendor/laravel/framework/src/Illuminate/Container/Container.php(633): Illuminate\\Container\\Container->build('GrahamCampbell\\\\...', Array)\n#3 /var/www/Cachet/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(697): Illuminate\\Container\\Container->make('GrahamCampbell\\\\...', Array)\n#4 /var/www/Cachet/vendor/laravel/framework/src/Illuminate/Container/Container.php(853): Illuminate\\Foun in /var/www/Cachet/vendor/laravel/framework/src/Illuminate/Container/Container.php on line 839
[Fri May 13 23:52:47.849128 2016] [:error] [pid 20386] [client blahblahblah:55327] PHP Fatal error:  Uncaught Illuminate\\Contracts\\Container\\BindingResolutionException: Unresolvable dependency resolving [Parameter #0 [ <required> $path ]] in class GrahamCampbell\\Exceptions\\ExceptionInfo in /var/www/Cachet/vendor/laravel/framework/src/Illuminate/Container/Container.php:839\nStack trace:\n#0 /var/www/Cachet/vendor/laravel/framework/src/Illuminate/Container/Container.php(806): Illuminate\\Container\\Container->resolveNonClass(Object(ReflectionParameter))\n#1 /var/www/Cachet/vendor/laravel/framework/src/Illuminate/Container/Container.php(777): Illuminate\\Container\\Container->getDependencies(Array, Array)\n#2 /var/www/Cachet/vendor/laravel/framework/src/Illuminate/Container/Container.php(633): Illuminate\\Container\\Container->build('GrahamCampbell\\\\...', Array)\n#3 /var/www/Cachet/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(697): Illuminate\\Container\\Container->make('GrahamCampbell\\\\...', Array)\n#4 /var/www/Cachet/vendor/laravel/framework/src/Illuminate/Container/Container.php(853): Illuminate\\Foun in /var/www/Cachet/vendor/laravel/framework/src/Illuminate/Container/Container.php on line 839
```
appears in the logs and I see a 500 error in my browser.

### Steps to reproduce

*If your issue requires any specific steps to reproduce, please outline them here.*

1. Clone repo @ tag v2.2.2 (git clone https://github.com/cachethq/Cachet.git -b v2.2.2)
2. Setup .env and run setup commands
3. Load page, and get error
4. `rm -rf bootstrap/cache/*`, `chmod 777 -R storage/` and `chmod 777 .env` to make sure that it's not a permissions issue

I've deleted this multiple times and redownloaded (even using your git checkout method) and still have this error. I have not tried another version yet.

**Comment by jbrooksuk** [comment]
Can I see your `.env` file, but be sure to redact the sensitive information :)

**Comment by awesomebing1** [comment]
Sure!

```
APP_ENV=production
APP_DEBUG=false
APP_URL=http://status.(redacted)
APP_KEY=base64:(redacted)

DB_DRIVER=mysql
DB_HOST=localhost
DB_DATABASE=cachet
DB_USERNAME=cachet
DB_PASSWORD=(redacted)
DB_PORT=3306
DB_PREFIX=cachet

CACHE_DRIVER=file
SESSION_DRIVER=file
QUEUE_DRIVER=sync
CACHET_EMOJI=false

MAIL_DRIVER=smtp
MAIL_HOST=mailtrap.io
MAIL_PORT=2525
MAIL_USERNAME=(redacted)
MAIL_PASSWORD=(redacted)
MAIL_ADDRESS=(redacted)@(redacted)
MAIL_NAME=null
MAIL_ENCRYPTION=tls

REDIS_HOST=localhost
REDIS_DATABASE=0
REDIS_PORT=6379

GITHUB_TOKEN=null
```

**Comment by GrahamCampbell** [comment]
Does anything of the stuff marked as "redacted" contain spaces? If so, you'll need to double quote them.

**Comment by GrahamCampbell** [comment]
When I say quote them, I mean wrap them in double quotes, not paste them here, lol. ;)

**Comment by awesomebing1** [comment]
No, nothing in this file contains spaces, unless a space at the end might have caused it?

**Comment by awesomebing1** [comment]
There's no spaces at the end of my variables, so that rules that theory out.

**Comment by GrahamCampbell** [comment]
It would appear that something is crashing the application before the exceptions handler service package has been register.

**Comment by awesomebing1** [comment]
At vendor/laravel/framework/src/Illuminate/Container/Container.php (in the root Cachet directory) right above L839, I added some debugging, changing the code to the following:

```
        $message = "Unresolvable dependency resolving [$parameter] in class {$parameter->getDeclaringClass()->getName()}";

        $e = new \Exception;
        var_dump($e->getTraceAsString());
        throw new BindingResolutionException($message);
``` 

That gave the following output:

```
string(2701) "#0 /var/www/Cachet/vendor/laravel/framework/src/Illuminate/Container/Container.php(806): Illuminate\Container\Container->resolveNonClass(Object(ReflectionParameter)) 
#1 /var/www/Cachet/vendor/laravel/framework/src/Illuminate/Container/Container.php(777): Illuminate\Container\Container->getDependencies(Array, Array) 
#2 /var/www/Cachet/vendor/laravel/framework/src/Illuminate/Container/Container.php(633): Illuminate\Container\Container->build('GrahamCampbell\\...', Array) 
#3 /var/www/Cachet/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(697): Illuminate\Container\Container->make('GrahamCampbell\\...', Array) 
#4 /var/www/Cachet/vendor/laravel/framework/src/Illuminate/Container/Container.php(855): Illuminate\Foundation\Application->make('GrahamCampbell\\...') 
#5 /var/www/Cachet/vendor/laravel/framework/src/Illuminate/Container/Container.php(808): Illuminate\Container\Container->resolveClass(Object(ReflectionParameter)) 
#6 /var/www/Cachet/vendor/laravel/framework/src/Illuminate/Container/Container.php(777): Illuminate\Container\Container->getDependencies(Array, Array) 
#7 /var/www/Cachet/vendor/laravel/framework/src/Illuminate/Container/Container.php(633): Illuminate\Container\Container->build('CachetHQ\\Cachet...', Array)
#8 /var/www/Cachet/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(697): Illuminate\Container\Container->make('CachetHQ\\Cachet...', Array) 
#9 /var/www/Cachet/vendor/graham-campbell/exceptions/src/ExceptionHandlerTrait.php(183): Illuminate\Foundation\Application->make('CachetHQ\\Cachet...') 
#10 /var/www/Cachet/vendor/graham-campbell/exceptions/src/ExceptionHandlerTrait.php(143): GrahamCampbell\Exceptions\ExceptionHandler->make(Array) 
#11 /var/www/Cachet/vendor/graham-campbell/exceptions/src/ExceptionHandlerTrait.php(112): GrahamCampbell\Exceptions\ExceptionHandler->getDisplayer(Object(Illuminate\Http\Request), Object(ErrorException), Object(ErrorException), 500) 
#12 /var/www/Cachet/vendor/graham-campbell/exceptions/src/ExceptionHandlerTrait.php(89): GrahamCampbell\Exceptions\ExceptionHandler->getResponse(Object(Illuminate\Http\Request), Object(ErrorException), Object(ErrorException)) 
#13 /var/www/Cachet/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(291): GrahamCampbell\Exceptions\ExceptionHandler->render(Object(Illuminate\Http\Request), Object(ErrorException)) 
#14 /var/www/Cachet/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(103): Illuminate\Foundation\Http\Kernel->renderException(Object(Illuminate\Http\Request), Object(ErrorException))
#15 /var/www/Cachet/public/index.php(54): Illuminate\Foundation\Http\Kernel->handle(Object(Illuminate\Http\Request))
#16 {main}" 
```

It looks like it can't load a class, however I don't use this framework so I don't know much about it.

**Comment by awesomebing1** [comment]
I finally figured out the issue, after an hour of debugging.

I redownloaded everything and finally figured out that I had to chmod 777 the cache directory. I then got
`[2016-05-14 16:19:14] production.ERROR: ErrorException: file_put_contents(/var/www/Cachet/.env): failed to open stream: Permission denied in /var/www/Cachet/app/Http/Controllers/SetupController.php:255` in the console. I then ran `sudo chmod 777 .env`, reloaded the page, and got the Setup page!

TL;DR: my permissions were messed up, try `chmod -R 777 cache`ing to see if it worked for you.
